pkg/scheduler/framework/types.go

```diff
@@ -80,9 +81,23 @@ type PodInfo struct {
 
 // AffinityTerm is a processed version of v1.PodAffinityTerm.
 type AffinityTerm struct {
-	Namespaces  sets.String
-	Selector    labels.Selector
-	TopologyKey string
+	Namespaces        sets.String
+	Selector          labels.Selector
+	TopologyKey       string
+	NamespaceSelector labels.Selector
+}
+
+// MergeNamespaces updates Namespaces with the set of namespaces identified by
+// NamespaceSelector.
+func (at *AffinityTerm) MergeNamespaces(nsLister listersv1.NamespaceLister) error {
+	ns, err := nsLister.List(at.NamespaceSelector)
+	if err != nil {
+		return err
+	}
+	for _, n := range ns {
+		at.Namespaces.Insert(n.Name)
```

Comment:
Even the NamespaceSelector excludes some namespace from Namespaces field, we still keep those namespace which in the Namespaces field. I guess that is what you mean about the union.

Reply:
oops, the KEP already said the support operator is limited, so this isn't a problem.